internal/service/componentdescriptor/service.go

```diff
@@ -0,0 +1,223 @@
+package componentdescriptor
+
+import (
+	"archive/tar"
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+
+	containerregistryv1 "github.com/google/go-containerregistry/pkg/v1"
+	"ocm.software/ocm/api/ocm/compdesc"
+	"ocm.software/ocm/api/ocm/extensions/repositories/genericocireg"
+
+	"github.com/kyma-project/lifecycle-manager/internal/descriptor/types"
+	"github.com/kyma-project/lifecycle-manager/internal/descriptor/types/ocmidentity"
+)
+
+const (
+	MaxDescriptorSizeBytes      = 100 * 1024 // 100KiB, our average is around 4KiB
+	TarReadChunkSize            = 10 * 1024  // 10KiB, for our average size we'll read it in one go
+	ComponentDescriptorFileName = compdesc.ComponentDescriptorFileName
+)
+
+var (
+	ErrInvalidArg       = errors.New("invalid argument")
+	ErrLayerNil         = errors.New("ComponentDescriptorLayer is nil in ComponentDescriptorConfig")
+	ErrLayerDigestEmpty = errors.New("ComponentDescriptorLayer.Digest is empty in ComponentDescriptorConfig")
+	ErrNotFoundInTar    = errors.New("not found in TAR archive")
+	ErrTarTooLarge      = errors.New("entry in the TAR archive is too large")
+	ErrLayerEmpty       = errors.New("layer content is empty")
+	ErrDecode           = errors.New("failed to decode component descriptor")
+)
+
+type OCIRepository interface {
+	GetConfigFile(ctx context.Context, name, tag string) ([]byte, error)
+	PullLayer(ctx context.Context, name, tag, digest string) (containerregistryv1.Layer, error)
+}
```

Comment:
How about passing the `ocmidentity.Component` directly instead of separated name and tag (version)

Reply:
Digest could also use the digest type. In the call below, the typecast should not be needed anyway.

Reply:
> How about passing the `ocmidentity.Component` directly instead of separated name and tag (version)

I was thinking about it, but it is "polluting" OCIRepository with an abstraction that doesn't belong there (OCM is not something that OCI Repository "knows" about). We could have a dedicated type that represents <name, tag, digest> and do the conversion. Because this repository is very simple, I decided for just strings. But I am always in favor of having a dedicated type.
Should I introduce it?
Or we could use the fact that the Component type offers methods: Name() and Version() - we could accept an interface that matches these methods.